x-pack/plugins/security_solution/server/usage/detections/detections_helpers.ts

```diff
@@ -301,3 +483,116 @@ export const getMlJobMetrics = async (
 
   return [];
 };
+
+export const getDetectionRuleMetrics = async (
+  kibanaIndex: string,
+  signalsIndex: string,
+  esClient: ElasticsearchClient,
+  savedObjectClient: SavedObjectsClientContract
+): Promise<DetectionRuleAdoption> => {
+  let rulesUsage: DetectionRulesTypeUsage = initialDetectionRulesUsage;
+  const ruleSearchOptions: RuleSearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: [],
+    ignoreUnavailable: true,
+    index: kibanaIndex,
+    size: 10_000,
+  };
+
+  let detectionAlertsResp: AlertsAggregationResponse | undefined;
+
+  try {
+    const { body: ruleResults } = await esClient.search<RuleSearchResult>(ruleSearchOptions);
+
+    // @ts-expect-error `SearchResponse['hits']['total']` incorrectly expects `number` type instead of `{ value: number }`.
+    ({ body: detectionAlertsResp } = await esClient.search({
+      index: signalsIndex,
```

Comment:
As frank mentioned above, `signalsIndex` from the config is isn't a concrete index nor an alias so I don't think this is going to work without a wildcard (but I could always be wrong 😅 ).